jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ResourceServlet.java

```diff
@@ -0,0 +1,929 @@
+ *   <dd>
+ *     The max size in bytes of the cache for static resources.
+ *     Use {@code -1} for the default value (256 MiB) or {@code 0} for no cache.
+ *   </dd>
+ *   <dt>otherGzipFileExtensions</dt>
+ *   <dd>
+ *     A comma-separated list of extensions of files whose content is implicitly
+ *     gzipped.
+ *     Defaults to {@code .svgz}.
+ *   </dd>
+ *   <dt>pathInfoOnly</dt>
+ *   <dd>
+ *     Use {@code true} to use only the pathInfo portion of a PATH (aka prefix) match.
+ *     Defaults to {@code true}.
+ *   </dd>
+ *   <dt>precompressed</dt>
+ *   <dd>
+ *     Omitted by default, so that no pre-compressed content will be served.
+ *     If set to {@code true}, the default set of pre-compressed formats will be used.
+ *     Otherwise can be set to a comma-separated list of {@code encoding=extension} pairs,
+ *     such as: {@code br=.br,gzip=.gz,bzip2=.bz}, where {@code encoding} is used as the
+ *     value for the {@code Content-Encoding} header.
+ *   </dd>
+ *   <dt>redirectWelcome</dt>
+ *   <dd>
+ *     Use {@code true} to redirect welcome files, otherwise they are forwarded.
+ *     Defaults to {@code false}.
+ *   </dd>
+ *   <dt>stylesheet</dt>
+ *   <dd>
+ *     Defaults to the {@code Server}'s default stylesheet, {@code jetty-dir.css}.
+ *     The path of a custom stylesheet to style the directory listing HTML.
+ *   </dd>
+ *   <dt>useFileMappedBuffer</dt>
+ *   <dd>
+ *     Use {@code true} to use file mapping to serve static resources.
+ *     Defaults to {@code false}.
+ *   </dd>
+ *   <dt>welcomeServlets</dt>
+ *   <dd>
+ *     Use {@code false} to only serve welcome resources from the file system.
+ *     Use {@code true} to dispatch welcome resources to a matching Servlet
+ *     (for example mapped to {@code *.welcome}), when the welcome resources
+ *     does not exist on file system.
+ *     Use {@code exact} to dispatch welcome resource to a Servlet whose mapping
+ *     is exactly the same as the welcome resource (for example {@code /index.welcome}),
+ *     when the welcome resources does not exist on file system.
+ *     Defaults to {@code false}.
+ *   </dd>
+ * </dl>
+ */
+public class ResourceServlet extends HttpServlet
+{
+    private static final Logger LOG = LoggerFactory.getLogger(ResourceServlet.class);
+
+    private ServletResourceService _resourceService;
+    private WelcomeServletMode _welcomeServletMode;
+    private boolean _pathInfoOnly;
+
+    public ResourceService getResourceService()
+    {
+        return _resourceService;
+    }
+
+    @Override
+    public void init() throws ServletException
+    {
+        ServletContextHandler contextHandler = initContextHandler(getServletContext());
+        _resourceService = new ServletResourceService(contextHandler);
+        _resourceService.setWelcomeFactory(_resourceService);
+        Resource baseResource = contextHandler.getBaseResource();
+
+        String rb = getInitParameter("baseResource", "resourceBase");
```

Comment:
I thought you said that the `ResourceServlet` isn't configurable by init params???

Reply:
see above